pkg/operator/reporting/templates_test.go

```diff
@@ -0,0 +1,569 @@
+				ReportDataSources: []*metering.ReportDataSource{ds1},
+				PrestoTables: []*metering.PrestoTable{
+					newTestPrestoTable("", testNamespace, testNamespace, testCatalogName, nil),
+				},
+			},
+			templateContext: TemplateContext{
+				Report: ReportTemplateInfo{
+					Inputs: map[string]interface{}{"UnreadyDeploymentReplicas": ""},
+				},
+			},
+			expectErr:    true,
+			expectErrMsg: "error executing template: template: reportQueryTemplate:1:17: executing \"reportQueryTemplate\" at <dataSourceTableName ...>: error calling dataSourceTableName: ReportDataSource  dependency not found",
+		},
+		{
+			name: "ReportDataSource dependencies were not found (due to no data sources) results in error",
+			reportTemplate: &ReportQueryTemplateContext{
+				ReportQuery:       newTestReportQuery(testReportName, testNamespace, ds1, "SELECT * FROM {| dataSourceTableName .Report.Inputs.UnreadyDeploymentReplicas |}"),
+				ReportDataSources: nil,
+				PrestoTables: []*metering.PrestoTable{
+					newTestPrestoTable("", testNamespace, testNamespace, testCatalogName, nil),
+				},
+			},
+			templateContext: TemplateContext{
+				Report: ReportTemplateInfo{
+					Inputs: map[string]interface{}{"UnreadyDeploymentReplicas": ""},
+				},
+			},
+			expectErr:    true,
+			expectErrMsg: fmt.Sprintf("error executing template: template: reportQueryTemplate:1:17: executing \"reportQueryTemplate\" at <dataSourceTableName ...>: error calling dataSourceTableName: ReportDataSource %s dependency not found", ""),
+		},
+		{
+			name: "ReportDataSource dependencies were not found (due to Report.Inputs referencing a ReportQuery) results in error",
+			reportTemplate: &ReportQueryTemplateContext{
+				ReportQuery:       newTestReportQuery(testReportName, testNamespace, ds1, "SELECT * FROM {| dataSourceTableName .Report.Inputs.UnreadyDeploymentReplicas |}"),
+				ReportDataSources: []*metering.ReportDataSource{ds1},
+				PrestoTables: []*metering.PrestoTable{
+					newTestPrestoTable(ds1.Status.TableRef.Name, testNamespace, testSchemaName, testCatalogName, nil),
+				},
+			},
+			templateContext: TemplateContext{
+				Report: ReportTemplateInfo{
+					Inputs: map[string]interface{}{"UnreadyDeploymentReplicas": testReportQuery},
+				},
+			},
+			expectErr:    true,
+			expectErrMsg: fmt.Sprintf("error executing template: template: reportQueryTemplate:1:17: executing \"reportQueryTemplate\" at <dataSourceTableName ...>: error calling dataSourceTableName: ReportDataSource %s dependency not found", testReportQuery),
+		},
+		{
+			name: "ReportDataSource dependencies were not found (due to Report.Inputs referencing an integer) results in error",
+			reportTemplate: &ReportQueryTemplateContext{
+				ReportQuery:       newTestReportQuery(testReportName, testNamespace, ds1, "SELECT * FROM {| dataSourceTableName .Report.Inputs.UnreadyDeploymentReplicas |}"),
```

Comment:
I was more looking for new queries that used these different types of inputs, and had different input definitions using these other types.

Reply:
Just added some more in the most recent commit. Still might be lacking some test cases using the time type, but without using `ReportingStart` and/or `ReportingEnd`.